File: src/se/fredsberg/game/zombiedice/TurnFactory.java
package se.fredsberg.game.zombiedice;

import java.util.HashMap;
import java.util.Map;

import se.fredsberg.game.container.Pool;
import se.fredsberg.game.dice.zombie.ZombieDie;
import se.fredsberg.game.dice.zombie.ZombieValue;
import se.fredsberg.game.random.RandomScrambler;

public class TurnFactory {

    public Turn createTurn() {
        ZombieDicePool cup = new ZombieDicePool();
        cup.shake(new RandomScrambler());
        return new Turn(cup, createPiles());
    }

    public Map<ZombieValue, Pool<ZombieDie>> createPiles() {
        Map<ZombieValue, Pool<ZombieDie>> piles = new HashMap<ZombieValue, Pool<ZombieDie>>();
        for (ZombieValue value : ZombieValue.values()) {
            piles.put(value, new Pool<ZombieDie>());
        }
        return piles;
    }

}

File: jssrc/TurnFactory.js
/**
 *
 */

function TurnFactory() {

	this.createPiles = function() {
		var piles = new Array(Object.keys(ZombieValue.values).length);
		for (var i = 0; i < piles.length; i++) {
			piles[i] = new Pool();
		}
		return piles;
	};

};

TurnFactory.prototype.createTurn = function() {
	var cup = new ZombieDicePool();
	cup.shake(new RandomScrambler());
	return new Turn(cup, this.createPiles());
};
